src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -46,13 +46,50 @@ class ActivityFeed extends React.Component<Props, State> {
 
     feedContainer: null | HTMLElement;
 
+    componentDidMount() {
+        this.resetFeedScroll();
+    }
+
+    componentDidUpdate(prevProps: Props, prevState: State) {
+        const {
+            currentUser: previousUser,
+            feedItems: prevFeedItems,
+        } = prevProps;
+        const { currentUser, feedItems: currFeedItems } = this.props;
+        const { isInputOpen: prevIsInputOpen } = prevState;
+        const { isInputOpen: currIsInputOpen } = this.state;
+
+        const hasMoreItems =
+            prevFeedItems &&
+            currFeedItems &&
+            prevFeedItems.length < currFeedItems.length;
+        const hasNewItems = !prevFeedItems && currFeedItems;
+        const hasNewUser = currentUser !== previousUser;
```

Comment:
Is this going from anonymous to logged in?

Reply:
The name for this is kind of confusing. The reason that I'm checking for the difference in user is I have a case where if the tab changes from ____ to Activity, I want to call resetScrollFeed(). I noticed that the prop that was changing was the user object (changing from undefined to defined) so I thought I would use that as an indicator of the tabs having changed

Reply:
Hmm, okay. I think using the user object as a proxy for tab change is a bit fragile. Since tabs are unmounted when switching from one to another, shouldn't `componentDidMount` catch the case you describe?